orte/mca/plm/rsh/plm_rsh_module.c

```diff
@@ -709,6 +713,38 @@ static int setup_launch(int *argcptr, char ***argvptr,
     return ORTE_SUCCESS;
 }
 
+/* close all open file descriptors w/ exception of stdin/stdout/stderr */
+static int close_open_file_descriptors(void) {
+    DIR *dir = opendir("/proc/self/fd");
```

Comment:
This won't work on non-Linux systems (e.g., MacOS).

Reply:
Is it safe to wrap function into `#ifdef __linux__`? Or should we check existence of `opendir` and `dirfd`?

Reply:
@jsquyres This is the same code used in the odls today, so I suspect we have a similar problem there. The issue here isn't a missing ```opendir```, but rather that the ```/proc``` directories aren't present. What we should do is fall back to the "slow" way of closing things instead of returning an error.